rust/operator-binary/src/superset_controller.rs

```diff
@@ -368,3 +504,109 @@ fn build_server_rolegroup_statefulset(
 pub fn error_policy(_error: &Error, _ctx: Context<Ctx>) -> Action {
     Action::requeue(Duration::from_secs(5))
 }
+
+pub fn append_authentication_volumes_and_volume_mounts(
+    authentication_class: &AuthenticationClass,
+    volumes: &mut Vec<Volume>,
+    volume_mounts: &mut Vec<VolumeMount>,
+) {
+    let authentication_class_name = authentication_class.metadata.name.as_ref().unwrap();
+
+    match &authentication_class.spec.provider {
+        AuthenticationClassProvider::Ldap(ldap) => {
+            if let Some(bind_credentials) = &ldap.bind_credentials {
+                let volume_name = format!("{authentication_class_name}-bind-credentials");
```

Comment:
Is there a reason that we don't hard-code the volume name to something like `ldap-bind-credentials?`

Reply:
Superset only supports a single auth method. Other products support multiple auth methods and therefore need multiple volumes (e.g. multiple LDAP credentials)
I would like to have consistent volumes across the products (that's the reason i put the get_volumes() in the operator-rs ;))
The easiest way to have non-clashing volume names seemed to me to prefix them with the auth class name

Reply:
I'd use an index or something in that case, any time we use values provided by the user we expose ourselves to Shenanigans:tm:.

Things get a bit muddy, but a person with permission to create a `SupersetCluster` doesn't necessarily have permission to create `Pod`s directly.

Reply:
In general I'm with you. But the user also can affect the name of created Pods and CMs and Services by naming the SupersetCluster object. In this case they can name the Volume name by naming the AuthenticationClass object.
I think a index is pretty complicated (it affects the volume mount, which path is needed at multiple occurrences) and worsens the debugging for the end-user when the LDAP connection does not work.

Reply:
Okay.